src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java

```diff
@@ -171,7 +243,10 @@ public void auditStarted(AuditEvent event) {
     @Override
     public void auditFinished(AuditEvent event) {
         String rendered = replaceVersionString(report);
-        rendered = rendered.replace(RESULTS_PLACEHOLDER, String.join(",\n", results));
+        rendered = rendered.replace(
+                RESULTS_PLACEHOLDER,
+                String.join(COMMA_LINE_SEPARATOR, results));
+        rendered = rendered.replace(RULES_PLACEHOLDER, generateRulesJson());
         writer.print(rendered);
```

Comment:
```suggestion
        writer.print(replaceVersionString(report).replace(
                RESULTS_PLACEHOLDER,
                String.join(COMMA_LINE_SEPARATOR, results)).replace(RULES_PLACEHOLDER, generateRulesJson()));
```
thats is whats happening.